ui/src/app/views/notebook-list/component.ts

```diff
@@ -158,6 +168,64 @@ export class NotebookListComponent implements OnInit, OnDestroy {
     this.openNotebook();
   }
 
+  renameThis(notebook: FileDetail): void {
+    this.notebookInFocus = notebook;
+    this.renameModal.open();
+  }
+
+  receiveRename($event: NotebookRename): void {
```

Comment:
nit: There doesn't seem to be a purpose for this method, can't you just call renameNotebook directly?